### PR TITLE
Add fixed product taxes support

### PR DESCRIPTION
**Pull request by supercid**

### CODE REVIEW

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 1]
```diff
@@ -151,6 +162,42 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
...
     }
 
+
+    /**
+     * In case merchant make use of Fixed Product Taxes (FPT)
+     * this function will add the value into the product list price
+     * @param Mage_Catalog_Model_Product $product
+     */
+    public function processWeeeTaxes(Mage_Catalog_Model_Product $product)
+    {
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: It should be handled in the Nosto_Object_Product_VariationCollection. And it should be handled together with the catalog price rules.

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 2]
```diff
@@ -151,6 +162,42 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
...
+        $weeeAttributekey = Mage::getModel('eav/entity_attribute')->getAttributeCodesByFrontendType('weee');
+        $weeeTaxes = $product->getData($weeeAttributekey[0]);
+
+        $variations = new Nosto_Object_Product_VariationCollection();
+        foreach ($storeAllowedCountriesList as $country) {
+            foreach ($weeeTaxes as $weeeTax) {
+                $variation = new Nosto_Tagging_Model_Meta_Variation();
+
+                if ($weeeTax['country'] === $country['value']) {
+                    $variation->setVariationId(strtoupper($country['label']));
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: What is in the label? May be the country code is better option

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 3]
```diff
@@ -151,6 +162,42 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
...
+                    $variation->setPrice(floatval($product->getPrice()));
+                    $variation->setListPrice($product->getPrice() + $weeeTax['value']);
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: I am not sure about it but please check should we handle the rounding here?

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 4]
```diff
@@ -151,6 +162,42 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
         return true;
     }
 
+
+    /**
+     * In case merchant make use of Fixed Product Taxes (FPT)
+     * this function will add the value into the product list price
+     * @param Mage_Catalog_Model_Product $product
+     */
+    public function processWeeeTaxes(Mage_Catalog_Model_Product $product)
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: Default variation id is not set

Comment (chenliangde) [reply app/code/community/Nosto/Tagging/Model/Meta/Product.php]: And the current variation id must be set to every pages and it must be one of the variation ids from the product

File: app/code/community/Nosto/Tagging/etc/config.xml [hunk 5]
```diff
@@ -28,7 +28,7 @@
 <config>
     <modules>
         <Nosto_Tagging>
-            <version>3.6.0</version>
+            <version>3.6.1</version>
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/etc/config.xml]: I think we should not update version number on the none release/hotfix PR

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 6]
```diff
@@ -140,6 +140,17 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
             $this->setCustomFields($this->loadCustomFields($product));
         }
 
+        // Price variations must be enabled in the config panel in order to use FPT
+        if (Mage::helper('weee')->isEnabled($store) &&
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: Does "Mage::helper('weee')" always return an object? Is the weee module a built-in module for enterprise version and community version?

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php (hunk 2 above)
Comment (supercid) [reply app/code/community/Nosto/Tagging/Model/Meta/Product.php]: It's the country name in the format ```FINLAND```. I'll switch to country code.

File: app/code/community/Nosto/Tagging/etc/config.xml (hunk 5 above)
Comment (supercid) [reply app/code/community/Nosto/Tagging/etc/config.xml]: Fixed after rebasing develop with master

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php (hunk 1 above)
Comment (supercid) [reply app/code/community/Nosto/Tagging/Model/Meta/Product.php]: Did you mean ```Nosto_Tagging_Model_Meta_Variation_Collection``` ?

Comment (chenliangde) [reply app/code/community/Nosto/Tagging/Model/Meta/Product.php]: Yes, you are right

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 7]
```diff
@@ -151,6 +162,42 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
...
+     */
+    public function processWeeeTaxes(Mage_Catalog_Model_Product $product)
+    {
+        $storeAllowedCountriesList = Mage::getModel('directory/country')
+            ->getResourceCollection()
+            ->loadByStore()
+            ->toOptionArray(true);
+
+        $weeeAttributekey = Mage::getModel('eav/entity_attribute')->getAttributeCodesByFrontendType('weee');
+        $weeeTaxes = $product->getData($weeeAttributekey[0]);
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: Add nullcheck for `$weeeAttributekey[0]`

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 8]
```diff
@@ -151,6 +162,42 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
...
+
+        $variations = new Nosto_Object_Product_VariationCollection();
+        foreach ($storeAllowedCountriesList as $country) {
+            foreach ($weeeTaxes as $weeeTax) {
+                $variation = new Nosto_Tagging_Model_Meta_Variation();
+
+                if ($weeeTax['country'] === $country['value']) {
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: Nullcheck for `$weeeTax['country']` ?

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php (hunk 3 above)
Comment (hannunosto) [reply app/code/community/Nosto/Tagging/Model/Meta/Product.php]: Perhaps nullcheck for `$weeeTax['value']`

**Pull request closed without merging**